### PR TITLE
Apama plugin documentation

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Apama plugin documentation

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/howto-guides/README.md [hunk 1]
```diff
@@ -14,3 +14,4 @@
 12. [How to connect an external device?](./013_connect_external_device.md)
 13. [How to access the logs on the device?](./014_thin_edge_logs.md)
 14. [How to install thin-edge.io on any Linux OS (no deb support)?](./015_installation_without_deb_support.md)
+15. [How to manage apama software artefatcs with apama plugin?](./016_apama_software_management_plugin.md)
```
Comment (rina23q) [inline docs/src/howto-guides/README.md:17]: @initard Just both should be aware, we have a conflict of file name `016_`. https://github.com/thin-edge/thin-edge.io/pull/703/files

File: docs/src/howto-guides/016_apama_software_management_plugin.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+```console
+Monitors
+========
+Name                                               Num Sub Monitors
+----                                               ----------------
+TedgeDemoMonitor                                             1
+```
+
+You can find more information on this apama project [here](https://github.com/thin-edge/thin-edge.io_examples/tree/main/StreamingAnalytics#testing-a-project)
```
Comment (rina23q) [inline docs/src/howto-guides/016_apama_software_management_plugin.md:77]:
```suggestion
You can find more information on this apama project [here](https://github.com/thin-edge/thin-edge.io_examples/tree/main/StreamingAnalytics#testing-a-project).
```

**Review by rina23q — APPROVED** [review]
Looks good. Just be careful about the conflict with the Alex's PR. If you merge first, no problem.

File: docs/src/howto-guides/016_apama_software_management_plugin.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Apama Software Management Plugin
+
+The apama plugin can be used to install and manage apama artefacts.
+This plugin can be used to install an apama project on a device as well as inject monitor files into an already installed project.
+
+
```
Comment (PradeepKiruvale) [inline docs/src/howto-guides/016_apama_software_management_plugin.md:6]: can remove some empty lines here

File: docs/src/howto-guides/016_apama_software_management_plugin.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Apama Software Management Plugin
+
```
Comment (PradeepKiruvale) [inline docs/src/howto-guides/016_apama_software_management_plugin.md:2]: isn't it better to mention that apama should be running on the system to use this plugin.

**Review by PradeepKiruvale — APPROVED** [review]
LGTM, some minor comments.

File: docs/src/howto-guides/016_apama_software_management_plugin.md (hunk 4 above)
Comment (albinsuresh) [reply docs/src/howto-guides/016_apama_software_management_plugin.md:2]: Apama need not be running. It just needs to be installed and it'll be started on the installation of a project and it'll be stopped when the project is removed.

**Pull request merged**